components/openzwave-adapter/src/id_map.rs

```diff
@@ -30,4 +30,11 @@ impl<Kind, Type> IdMap<Kind, Type> where Type: Eq + Clone, Kind: Clone {
         let find_result = guard.iter().find(|&&(ref id, _)| id == needle);
         find_result.map(|&(_, ref ozw_object)| ozw_object.clone())
     }
+
+    pub fn remove_by_ozw(&mut self, needle: &Type) {
+        let mut guard = self.map.write().unwrap(); // we have bigger problems if we're poisoned
+        if let Some(index) = guard.iter().position(|&(_, ref item)| item == needle) {
+            guard.remove(index);
+        }
+    }
```

Comment:
I think it makes more sense to just return the id (rather than the entire tuple).